Tasks/DeployAzureResourceGroup/operations/MachineGroupExtensionHelper.ts

```diff
@@ -0,0 +1,250 @@
+        return new Promise((resolve, reject) => {
+            var invokeCreateOrUpdate = (error, result, request, response) => {
+                if (error) {
+                    this.log(tl.loc("VMStartFailedSkipExtensionOperation", vmName, operationParameters.operation, extensionName));
+                    var callback = this._getPostOperationCallBack(operationParameters, extensionName, vmName, resolve, reject);
+                    callback("error");
+                }
+                else if (result) {
+                    var callback = this._getPostOperationCallBack(operationParameters, extensionName, vmName, resolve, reject);
+                    this._computeClient.virtualMachineExtensions.createOrUpdate(this._taskParameters.resourceGroupName, vmName, extensionName, parameters, callback);
+                }
+            }
+            this._computeClient.virtualMachines.start(this._taskParameters.resourceGroupName, vmName, invokeCreateOrUpdate);
+        });
+    }
+
+    private installExtensionOnRunningVm(operationParameters, vmName, extensionName, parameters): Promise<any> {
+        return new Promise((resolve, reject) => {
+            var callback = this._getPostOperationCallBack(operationParameters, extensionName, vmName, resolve, reject);
+            this._computeClient.virtualMachineExtensions.createOrUpdate(this._taskParameters.resourceGroupName, vmName, extensionName, parameters, callback);
+        })
+    }
+
+    private _getPostOperationCallBack(operationParameters, extensionName, vmName, resolve, reject) {
+        var postOperationCallBack = (error, result?, request?, response?) => {
+            if (error) {
+                operationParameters.failureCount++;
+                operationParameters.errors += error.message;
+                operationParameters.errors += "\n";
+                var msg = tl.loc("OperationFailed", operationParameters.operation, extensionName, vmName);
+                this.log(msg);
+                resolve(msg);
+            } else {
+                operationParameters.successCount++;
+                msg = tl.loc("OperationSucceeded", operationParameters.operation, extensionName, vmName);
+                this.log(msg);
+                resolve(msg);
+            }
+        }
+        return postOperationCallBack;
+    }
+
+    private getVmInstanceView(resourceGroupName, vmName, object, operationParameters): Promise<any> {
+        return new Promise((resolve, reject) => {
+            var getVmInstanceViewCallback = (error, result, request, response) => {
+                if (result) {
+                    resolve(result);
+                }
+                else if (error) {
+                    var errMsg = tl.loc("VMDetailsFetchFailedSkipExtensionOperation", vmName, operationParameters.operation);
+                    this.log(errMsg);
+                    resolve(errMsg);
```

Comment:
Should this be reject? if you are going to reject this, then dont ignore the original error and append it to the message using utils.getError(error)